File: sem4/MPP/mpp-proiect-repository-mariaruncan/ProjJavaFX/Model/src/main/java/app/model/dto/EmployeeDto.java
package app.model.dto;

import java.io.Serializable;

public class EmployeeDto implements Serializable {
    private String username;
    private String password;

    public EmployeeDto(String username, String password){
        this.username = username;
        this.password = password;
    }

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    @Override
    public String toString() {
        return "EmployeeDto{" +
                "username='" + username + '\'' +
                ", password='" + password + '\'' +
                '}';
    }
}

File: sem4/MPP/mpp-proiect-repository-mariaruncan/ProjCSharp/Model/dto/EmployeeDto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Model.dto
{
    [Serializable]
    public class EmployeeDto
    {
        public string Username { get; set; }
        public string Password { get; set; }

        public EmployeeDto(string username, string password)
        {
            Username = username;
            Password = password;
        }

        public override bool Equals(object obj)
        {
            return obj is EmployeeDto dto &&
                   Username == dto.Username &&
                   Password == dto.Password;
        }

        public override int GetHashCode()
        {
            int hashCode = 568732665;
            hashCode = hashCode * -1521134295 + EqualityComparer<string>.Default.GetHashCode(Username);
            hashCode = hashCode * -1521134295 + EqualityComparer<string>.Default.GetHashCode(Password);
            return hashCode;
        }
    }
}
